src/games/strategy/engine/framework/ui/NewGameChooserEntry.java

```diff
@@ -51,31 +59,34 @@ public NewGameChooserEntry(final URI uri)
   }
 
   public void fullyParseGameData() throws GameParseException {
+    // TODO: We should be setting this in the the constructor. At this point, you have to call methods in the
+    // correct order for things to work, and that is bads.
     m_data = null;
 
-    String error = null;
     final AtomicReference<String> gameName = new AtomicReference<>();
 
-    try (InputStream input = m_url.toURL().openStream()) {
+    Optional<InputStream> inputStream = UrlStreams.openStream(m_url);
+    if(!inputStream.isPresent()) {
+      return;
+    }
+
+    try (InputStream input = inputStream.get()) {
       m_data = new GameParser().parse(input, gameName, false);
       m_gameDataFullyLoaded = true;
 
     } catch (final EngineVersionException e) {
       ClientLogger.logQuietly(e);
-      error = e.getMessage();
+      throw new GameParseException(e.getMessage());
```

Comment:
here we just simplify. error is just used to throw a game parse exception. So we'll repeat the 'throw new' code a few times to avoid repeating 'error = e.getMessage()' and avoid the if statement at the bottom that checks if `error != null`